core/container.go

```diff
@@ -809,7 +867,7 @@ func (container *Container) Exec(ctx context.Context, gw bkgw.Client, defaultPla
 	// create /dagger mount point for the shim to write to
 	runOpts = append(runOpts,
 		llb.AddMount(metaMountDestPath,
-			llb.Scratch().File(meta),
+			llb.Scratch().File(meta, CustomName{Name: "creating dagger metadata", Internal: true}.LLBOpt(), payload.Group.LLBOpt()),
```

Comment:
@vito @sipsma Thoughts on this?

Basically serializing vertex names as JSON with `name` / `internal` / `group` (see previous comment)

Consumer is responsible for dealing with this. If the message fails JSON deserialization then assume it's `{name: <original payload>}`

Reply:
Definitely a bit hokey but I don't know a better alternative. Seems like something we could replace with upstream improvements in Buildkit (ability to marking things 'internal' or hidden by default, ability to set a ProgressGroup for resolve) -  but I wouldn't block on it.

Reply:
@vito Yep, indeed. BuildKit does support arbitrary key values (`WithCustomName` maps to `"llb.customname": name`), however those things don't make it back as solve status, AFAIK. Unless I'm mistaken?